### ISSUE TITLE
how to dereference swig double* returned by mfem.Vector().GetData() ?

### ISSUE THREAD

**Issue by michael-barrow-llnl** [mfem/PyMFEM#185]
TL;DR: I need to take the return of mfem.Vector().GetData() into a pybind11 wrapped method and can't get it to recognize SWIG's 'double *'

Hi, I need to dereference the 'double *' object returned by the mfem.Vector class GetData() method.
It seems I have two options to get at the data, but these are undesirable
1: Just do mfem.Vector()[0]
2: Use ctypes to access the address returned by mfem.Vector().GetData().

Is there some way to incorporate typemaps into the interface for raw pointer dereference or maybe the numpy interface to wrap the pointer?

I want to use the GetData() method in a pybind11 wrapped class and can't figure out how to define a signature type for the SWIG double*.

Cheers

**Comment by sshiraiwa** [comment]
Hi @michael-barrow-llnl, I am not sure about pybind11 wrapped class. Does it take a numpy array as input? If so, you can
use GetDataArray() instead of GetData(). This method returns numpy array which points to the same memory address of the original data.

```
v = mfem.Vector(np.array([1,2,3.]))
v
<mfem._ser.vector.Vector; proxy of <Swig Object of type 'mfem::Vector *' at 0x7ff827854480> >
v.Print()   -> it prints 1,2, 3
data = v.GetDataArray()
data[2] = 100
v.Print()   -> it prints 1,2, 100
```

**Comment by sshiraiwa** [comment]
Closing this for now. Feel free to reopen it.
